src/mad_spark_alt/core/answer_extractor.py

```
@@ -0,0 +1,405 @@
+        topic = re.sub(r'^(what|how|why|when|where|who)\s+(are|is|to|can|do|does)\s*', '', question.lower())
+        topic = re.sub(r'\b\d+\s+(ways?|methods?|steps?)\s*', '', topic)
+        topic = re.sub(r'\?', '', topic)
+        topic = topic.strip()
+        
+        # Take first meaningful phrase
+        if len(topic) > 50:
+            topic = topic[:50] + "..."
+        
+        return topic or "the challenge"
+    
+    def _convert_question_to_action(self, qadi_content: str, topic: str, number: int) -> str:
```

Ruff (0.11.9)

Unused method argument: `qadi_content`

(ARG002)